src/psyclone/psyGen.py

```diff
@@ -1739,9 +1776,124 @@ def dag_name(self):
         '''
         return "ACC_directive_" + str(self.abs_position)
 
+    def _add_region(self, start_text, end_text=None, data_movement=None):
```

Comment:
pylint issues here ...

W:1830,15: Access to a protected member _ast of a client class (protected-access)
W:1830,39: Access to a protected member _ast of a client class (protected-access)
W:1833,20: Access to a protected member _ast of a client class (protected-access)
W:1838,39: Access to a protected member _ast of a client class (protected-access)
W:1842,45: Access to a protected member _ast_end of a client class (protected-access)
W:1845,45: Access to a protected member _ast of a client class (protected-access)
W:1854,12: Access to a protected member _parent of a client class (protected-access)
W:1889, 8: Access to a protected member _parent of a client class (protected-access)
R:1779, 4: Too many branches (14/12) (too-many-branches)
W:1892, 8: Attribute '_ast_start' defined outside __init__ (attribute-defined-outside-init)
W:1857,12: Attribute '_ast_end' defined outside __init__ (attribute-defined-outside-init)

Reply:
Oh yes. It turns out that we only ever assign to _ast_start so I've got rid of that entirely and added getter methods for _ast and _ast_end. I'm leaving the _parent refs as they are already covered by the TODO (fparser/#102).